nats.go

```diff
@@ -1542,6 +1542,8 @@ func (s *Subscription) NextMsg(timeout time.Duration) (msg *Msg, err error) {
 			return nil, ErrConnectionClosed
 		}
 		delivered := atomic.AddUint64(&s.delivered, 1)
+		// FIXME: jam: off to the races on max vs. delivered
+		// FIXME: jam: cached above but possibly updated...
 		if max > 0 {
 			if delivered > max {
 				return nil, errors.New("nats: Max messages delivered")
```

Comment:
Just commenting here on other race conditions. We do not want to check delivered/etc. in all these different ways. Longer term fix to talk through/discuss with test cases.